src/ext/java/org/opentripplanner/ext/reportapi/resource/ReportResource.java

```diff
@@ -62,4 +71,20 @@ public Response getBicycleSafetyAsCsv(
       )
       .build();
   }
+
+  @GET
+  @Path("/graph.json")
+  public Response stats(@Context OtpServerRequestContext serverRequestContext) {
+    // since the computation is pretty expensive only allow it every 5 minutes
+    if (cachedStats == null) {
+      Supplier<GraphStats> getStats = () -> GraphReportBuilder.build(serverRequestContext);
+      cachedStats = new CachedValue<>(getStats, Duration.ofMinutes(5));
+    }
```

Comment:
To make this thread safe, you must init it it like this:
```Java
    // since the computation is pretty expensive only allow it every 5 minutes
    private static CachedValue<GraphStats> cachedStats = new CachedValue<>(
        () -> GraphReportBuilder.build(serverRequestContext),
        Duration.ofMinutes(5)
    );
```
Then we need to change the cache a little.

Reply:
The supplier must be changed so it does not call the supplier before it is needed - so we need to lazy-initialized the value. Sorry, I missed this part the first time around.

Reply:
If we make it static then we don't have the serverRequestContext. How should this work in this case?

Reply:
I do not think there is a better option here, there should be a report service, witch does the cacheing but that is a much bigger refactoring.